src/core/build.jl

```diff
@@ -36,7 +36,7 @@ function _vars_type(md::ModelDef, comp_def::ComponentDef)
     var_defs = variables(comp_def)    
     vnames = Tuple([name(vdef) for vdef in var_defs])
     
-    first = comp_def.first
+    first = first_period(comp_def, md)
```

Comment:
Pretty much all the other functions that take a `Model`, `ModelInstance` or `ModelDef` have this as the first argument, so I'd make this `first_period(md, comp_def)` for consistency.